ui/lib/kv/addon/routes.js

```diff
@@ -6,14 +6,15 @@
 import buildRoutes from 'ember-engines/routes';
 
 export default buildRoutes(function () {
-  // There are two list routes because Ember won't let a query param (e.g. *path_to_secret) be blank.
+  // There are two list routes because Ember won't let a route param (e.g. *path_to_secret) be blank.
   // :path_to_secret is used when we're listing a secret directory. Example { path: '/:beep%2Fboop%2F/directory' });
   this.route('list');
   this.route('list-directory', { path: '/:path_to_secret/directory' });
   this.route('create');
   this.route('secret', { path: '/:name' }, function () {
-    this.route('details');
-    this.route('edit');
+    this.route('details', function () {
+      this.route('edit'); // route to create new version of a secret
```

Comment:
this makes the url something like `/my-kv-engine/kv/my-secret/details/edit?version=23` which I don't love as you're not exactly editing version 23 but creating a new version based off of 23

It's not very RESTful, but thoughts on doing something like the following which would yield: `/details/create-from?version=23` or maybe i'm overcomplicating 🙃 
```suggestion
      this.route('edit', { path: '/create-from' }); // route to create new version of a secret
```

Reply:
IMO: 💙. I think it could be confusing for future devs—without having their heads in KV for weeks—to follow the routing when they're seeing "edit" but we're never editing a secret. This clarifies what it is we're doing to both users and devs.

Reply:
I would prefer to see the route structure changed here so that versions have their own route that replaces the details route with a dynamic segment for the version number:
```js
this.route('versions', function () {
  this.route('create');
  this.route('version', { path: '/:version_number' }, function () {
    this.route('details');
  });
});
```
The url would then look something like `/my-kv-engine/kv/my-secret/versions/create` or `/my-kv-engine/kv/my-secret/versions/23/details`. To me, this more accurately describes what is being created and viewed. I understand this may impact scope so take it purely as a suggestion or something to consider in future iterations 😄.

Reply:
I agree this routing is more intuitive looking. The initial query for `kv/secret/data` doesn't have a version so this would require some extra logic on how we'd show the details without that route param (which is also missing if a user doesn't have metadata). Adding a link to this suggestion for me to come back to once I'm handling the permissions side of things